src/include/migraphx/set_operator.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_SET_OPERATOR_IMPL_HPP
+#define MIGRAPHX_GUARD_RTGLIB_SET_OPERATOR_IMPL_HPP
+
+#include <utility>
+namespace migraphx {
+namespace set_op {
```

Comment:
I dont think we need a separate namespace for this, but this is missing `inline namespace MIGRAPHX_INLINE_NS`.